connexion/decorators/TracerDecorator.py

```diff
@@ -0,0 +1,40 @@
+from ..utils import get_tracer
+import logging
+
+logger = logging.getLogger('')
+
+
+def TracerDecorator(func):
+    def wrapper(cls, response, mimetype=None, request=None):
+        tracer = get_tracer()
+
+        # if tracer is configured, then start a span now
+        if tracer is not None:
+            from opentracing.ext import tags
+            from opentracing.propagation import Format
+
+            # extract the context from request header to continue a session
+            # taken from https://github.com/yurishkuro/opentracing-tutorial/tree/master/python/lesson03#extract-the-span-context-from-the-incoming-request-using-tracerextract
+            if request is not None:
+                span_ctx = tracer.extract(Format.HTTP_HEADERS, request.headers)
+                span_tags = {tags.SPAN_KIND: tags.SPAN_KIND_RPC_SERVER}
+
+                # remove domain from url, so only the path is in the span
+                from urllib.parse import urlparse
+                path = urlparse(request.url).path
+
+                scope = tracer.start_span(path + "_" + request.method, child_of=span_ctx, tags=span_tags)
+                scope.log_kv({"request": request})
+            else:
+                scope = tracer.start_span("TracerDecorator")
+
+            resp = func(cls, response, mimetype, request)
+
+            scope.log_kv({"response": response})
```

Comment:
are we working under the assumption that these calls do not throw? as if they do we will fail a successful response? do we need a try-except-finally here?

Reply:
The variable func holds the function: get_response from flask_api.py ([L#120](https://github.com/zalando/connexion/blob/13809da779dfddc474bcfe38541983c9a5aa2a5d/connexion/apis/flask_api.py#L120), see your other comment).
This function returns a ConnexionRequest-Object. So it doesn't throw an error in my understanding.